lib/blockscore/collection.rb

```diff
@@ -1,18 +1,239 @@
+      assoc_params = default_params.merge(params)
+      instance = member_class.create(assoc_params)
+
+      new_member(instance) do |member|
+        register_to_parent(member)
+      end
+    end
+
+    # Retrieve a collection member by its id
+    #
+    # @example usage
+    #   person.question_sets.retrieve('55ef5b4e3532630003000178') # => instance of QuestionSet
+    #
+    # @param id [String] resource id
+    #
+    # @return instance of {member_class} if found
+    # @raise [BlockScore::NotFoundError] otherwise
+    #
+    # @api public
+    def retrieve(id)
+      each do |item|
+        next unless item.id == id
+        return item
+      end
+
+      instance = member_class.retrieve(id)
+
+      new_member(instance) do |member|
+        register_to_parent(member)
+      end
+    end
+
+    protected
+
+    # @!attribute [r] member_class
+    # class which will be used for the embedded
+    # resources in the collection
+    #
+    # @return [Class]
+    #
+    # @api private
+    attr_reader :member_class
+
+    # Default params for making an instance of {member_class}
+    #
+    # @return [Hash]
+    #
+    # @api private
+    def default_params
+      {
+        foriegn_key => parent.id
+      }
     end
 
     private
 
-    def method_missing(method, *args, &block)
-      @target.public_send(method, *args, &block)
+    # Generate foriegn key name for parent resource
+    #
+    # @return [Symbol] resource name as id
+    #
+    # @api private
+    def foriegn_key
+      :"#{parent_name}_id"
+    end
+
+    # Initialize a new collection member
+    #
+    # @param instance [BlockScore::Base] collection member instance
+    # @yield [Member] initialized member
+    #
+    # @return [Member] new member
+    #
+    # @api private
+    def new_member(instance, &blk)
+      Member.new(parent, instance).tap(&blk)
+    end
+
+    # Check if `parent_id` is defined on `item`
+    #
+    # @param item [BlockScore::Base] any resource
+    #
+    # @return [Boolean]
+    #
+    # @api private
+    def parent_id?(item)
+      parent.id && item.send(foriegn_key) == parent.id
+    end
+
+    # Register a resource in collection
+    #
+    # @param item [BlockScore::Base] a resource
+    #
+    # @raise [BlockScore::Error] if no `parent_id`
+    # @return [BlockScore::Base] otherwise
+    #
+    # @api private
+    def register_to_parent(item)
+      fail Error, 'None belonging' unless parent_id?(item)
+      ids << item.id
+      self << item
+      item
+    end
+
+    # Fetches embedded ids from parent and adds to self
+    #
+    # @return [undefined]
+    #
+    # @api private
+    def register_parent_data
+      ids.each do |id|
+        item = member_class.retrieve(id)
+        self << item
+      end
+    end
+
+    # ids that belong to the collection
+    #
+    # @return [Array<String>]
+    #
+    # @api private
+    def ids
+      parent.attributes.fetch(:"#{Util.to_plural(member_class.resource)}", [])
```

Comment:
Looking at this again, I'm thinking `Util.to_plural(member_class.resource)` should just be its own method